functions/onprem/orborus/orborus.go

```diff
@@ -1404,10 +1445,119 @@ func main() {
 	//defer cleanup()
 
 	// Block until a signal is received
-	if isRunningInCluster() {
+	if shuffle.IsRunningInCluster() {
 		log.Printf("[INFO] Running inside k8s cluster")
 	}
 
+	if isKubernetes == "true" {
+		clientset, _, err := shuffle.GetKubernetesClient()
+		if err != nil {
+			log.Printf("[ERROR] Error getting kubernetes client:", err)
+			os.Exit(1)
+		}
+
+		kubernetesNamespace := "default"
+
+		// Check if namespace exist as variable. If so, make it
+		if len(os.Getenv("KUBERNETES_NAMESPACE")) > 0 && !namespacemade {
+			kubernetesNamespace = os.Getenv("KUBERNETES_NAMESPACE")
+		}
+
+		// fix roles
+		// check if "service-creator" role is assigned to the service account "default"
+		roleBindingName := "service-creator-binding"
+		serviceAccountName := "default"
+		// Check if the RoleBinding exists
+		roleBinding, err := clientset.RbacV1().RoleBindings(kubernetesNamespace).Get(context.TODO(), roleBindingName, metav1.GetOptions{})
+		if err != nil {
+			log.Printf("[WARNING] Failed to get RoleBinding %s: %s", roleBindingName, err)
```

Comment:
What happens in this err != nil seems like it should be a function of its own if all it does is create the right service account

Reply:
errors in creating a container. i took the "hope it fixes itself" strategy here on first draft lol (hope customer realises that they need a way to create these layers, through docs or support etc)

Reply:
I think these should always be controlled through the original setup for Orborus, then replicated across Worker & Apps, meaning they control it in their Orborus Yaml file. If it's not there, we prolly shouldn't force it 🤔 

PS: I also noticed the "context.TODO()" - can you do .Background() before we forget? TODO's never get read or rewritten